ts/src/base/Exchange.ts

```diff
@@ -864,7 +864,15 @@ export default class Exchange {
             maxCapacity: 1000,
             refillRate: (this.rateLimit > 0) ? 1 / this.rateLimit : Number.MAX_VALUE,
         }, this.tokenBucket);
-        this.throttler = new Throttler (this.tokenBucket);
+        this.throttler = this.throttlerFactory ({
+            'tokenBucket': this.tokenBucket,
+            'id': this.id,
+        });
+    }
+
+    throttlerFactory (opts: any) {
+        const tokenBucket = this.safeValue (opts, 'tokenBucket');
+        return new Throttler (tokenBucket);
     }
```

Comment:
In monorepo PR you pass a `throttlerFactory` function to ccxt. Are you sure it's going to be executed?

Reply:
The throttlerFactory method you can see here is the default one and can be overridden by passing the method in exchange parameters. 
It won't be executed if the `sharedThrottler` flag is true because we want to pass our factory and keep the shared instance of the throttler in the SDK.
Cf
```
 ...(sharedThrottler && { throttlerFactory: this.sharedThrottlerFactory.bind(this) }),
```
If this flag is false, it will use the default factory presented above.